--- a/scripts/toy.groovy
+++ b/scripts/toy.groovy
@@ -421,7 +421,7 @@
 						if (!cond || cond()) opts.add(r);
 					}
 					opts.push([ lbl: "Back", act: null ]);
-					def opt = getSelectedValue(stateIs("DEBUG") ? DOMME : "Yes, toy?", opts.collect { it.lbl });
+					def opt = getSelectedValue("Yes, toy?", opts.collect { it.lbl });
 					def act = opts[opt].act;
 					if (act)
 						act(opts[opt].arg);
