--- a/scripts/toy__sha256_0d24d6c78bca.groovy
+++ b/scripts/toy__sha256_0d24d6c78bca.groovy
@@ -67,18 +67,17 @@
 	final SPANKING_IMPLEMENTS = [CROP, PADDLE, WHIP, SPOON, CANE, RULER, HAIRBRUSH];
 	final LEVELS = [TEASE, DENIAL, KINK];
 	final getDomme = { domme, set = null ->
-		final readDomme = { path ->
-			if (!path) return [:];
-			final f = new File(path);
-			if (!f.exists()) return [:];
-			return Eval.me(f.text);
-		};
-		def d = [
-			"$DATAFOLDER/images/toy/domme.groovy",
-			"$DATAFOLDER/images/toy/$domme/person.groovy",
-			"$DATAFOLDER/images/toy/$domme/$set/set.groovy" ]
-				.collect { readDomme(it) }
-				.sum();
+		def d = [:];
+		d.put('title', 'Mistress');
+		if (domme == 'ancilla') {
+			d.put('name', 'Ancilla');
+			d.put('fullName', 'Ancilla Tilia');
+		}
+		else if (domme == 'emily') {
+			d.put('name', 'Emily');
+			d.put('fullName', 'Emily Marilyn');
+			d.put('locale', 'en-US');
+		}
 		d.id = domme;
 		return d;
 	};
--- a/scripts/toy.groovy
+++ b/scripts/toy.groovy
@@ -64,18 +64,17 @@
 	final SPANKING_IMPLEMENTS = [CROP, PADDLE, WHIP, SPOON, CANE, RULER, HAIRBRUSH];
 	final LEVELS = [TEASE, DENIAL, KINK];
 	final getDomme = { domme, set = null ->
-		final readDomme = { path ->
-			if (!path) return [:];
-			final f = new File(path);
-			if (!f.exists()) return [:];
-			return Eval.me(f.text);
-		};
-		def d = [
-			"$DATAFOLDER/images/toy/domme.groovy",
-			"$DATAFOLDER/images/toy/$domme/person.groovy",
-			"$DATAFOLDER/images/toy/$domme/$set/set.groovy" ]
-				.collect { readDomme(it) }
-				.sum();
+		def d = [:];
+		d.put('title', 'Mistress');
+		if (domme == 'ancilla') {
+			d.put('name', 'Ancilla');
+			d.put('fullName', 'Ancilla Tilia');
+		}
+		else if (domme == 'emily') {
+			d.put('name', 'Emily');
+			d.put('fullName', 'Emily Marilyn');
+			d.put('locale', 'en-US');
+		}
 		d.id = domme;
 		return d;
 	};
--- a/scripts/toy__sha256_bf0bc1679772.groovy
+++ b/scripts/toy__sha256_bf0bc1679772.groovy
@@ -47,18 +47,17 @@
 	final DRESSED = "dressed", TITS = "tits", PUSSY = "pussy", LINGERIE = "lingerie", TEASE = "tease", SIT = "sit", BOOTS = "boots", KNEEL = "kneel", STOOD = "stood", MEAN = "mean", CROP = "crop", ASS = "ass", SQUAT = "squat", SSSH = "sssh";
 	final nDRESSED = "!$DRESSED", nTITS = "!$TITS", nTEASE = "!$TEASE", nKNEEL = "!$KNEEL", nSQUAT = "!$SQUAT";
 	final getDomme = { domme, set = null ->
-		final readDomme = { path ->
-			if (!path) return [:];
-			final f = new File(path);
-			if (!f.exists()) return [:];
-			return Eval.me(f.text);
-		};
-		def d = [
-			"$DATAFOLDER/images/toy/domme.groovy",
-			"$DATAFOLDER/images/toy/$domme/person.groovy",
-			"$DATAFOLDER/images/toy/$domme/$set/set.groovy" ]
-				.collect { readDomme(it) }
-				.sum();
+		def d = [:];
+		d.put('title', 'Mistress');
+		if (domme == 'ancilla') {
+			d.put('name', 'Ancilla');
+			d.put('fullName', 'Ancilla Tilia');
+		}
+		else if (domme == 'emily') {
+			d.put('name', 'Emily');
+			d.put('fullName', 'Emily Marilyn');
+			d.put('locale', 'en-US');
+		}
 		d.id = domme;
 		return d;
 	};
