--- a/scripts/toy__sha256_0d24d6c78bca.groovy
+++ b/scripts/toy__sha256_0d24d6c78bca.groovy
@@ -189,8 +189,8 @@
 	void pause(BigDecimal t) { wait(t); }
 	// Random integer number, constant for play session or debug
 	int getRandomSession(int n) {
-		Random r = new Random(loadInteger("toy.play.started")?:loadInteger("toy.launch.lasttime"))
-		return r.nextInt(n);
+		final seed = loadInteger("toy.play.started")?:loadInteger("toy.launch.lasttime")?:0
+		return (seed % 2147483647) * 16807 % 2147483647 % n;
 	}
 	final adjustPunish = { p ->
 		if (p > 0) adjustChastePts((int)p);
--- a/scripts/toy.groovy
+++ b/scripts/toy.groovy
@@ -186,8 +186,8 @@
 	void pause(BigDecimal t) { wait(t); }
 	// Random integer number, constant for play session or debug
 	int getRandomSession(int n) {
-		Random r = new Random(loadInteger("toy.play.started")?:loadInteger("toy.launch.lasttime"))
-		return r.nextInt(n);
+		final seed = loadInteger("toy.play.started")?:loadInteger("toy.launch.lasttime")?:0
+		return (seed % 2147483647) * 16807 % 2147483647 % n;
 	}
 	final adjustPunish = { p ->
 		if (p > 0) adjustChastePts((int)p);
