--- a/scripts/toy.groovy
+++ b/scripts/toy.groovy
@@ -395,7 +395,7 @@
 	final getBooleanG = { msg, yes=null, no=null, shuffle=false ->
 		def yesText = yes?:gagText("Yes, ${dommeTitle()}", "yes");
 		def noText = no?:gagText("No, ${dommeTitle()}", "no");
-		shuffle = shuffle ? getRandom(2) : false
+		shuffle = shuffle ? getRandom(2) != 0 : false
 		def ans = shuffle ? !getBoolean(msg, noText, yesText) : getBoolean(msg, yesText, noText);
 		if (stateIs(GAGGED) && !getRandom(10)) {
 			speak(compose([["What was that?", "What?", "What did you say?", "Huh?", "Speak clearly!"]]));
