--- a/scripts/toy/lines.groovy
+++ b/scripts/toy/lines.groovy
@@ -463,7 +463,7 @@
 			showButtonGT("Hands cuffed, ${dommeTitle()}", "cuffed", 20, 1);
 		}
 		snapFinger();
-		present([nDRESSED, [TITS,PUSSY,ASS]], [
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [
 				["Here we go...", "Then let's start!", "Get ready!"]]);
 		pause(3);
 		def lines = prepareLines();
@@ -479,7 +479,7 @@
 			while (linenr==lastline) linenr = getRandom(lines.size());
 			line = compose(lines[linenr]).capitalize();
 			lastline = linenr;
-			cpm = writeLine([nDRESSED, [TITS,PUSSY,ASS]], line);
+			cpm = writeLine([nDRESSED, "$TITS|$PUSSY|$ASS"], line);
 			if (Math.abs(cpm) > cpmGoal*2.5) {
 				present(null, [["No way", "How comes"], ["you can"], ["suddenly", ""], 
 						["type", "write"], ["that fast!?"], ["Stop cheating", "Don't try to cheat"], 
--- a/scripts/toy/orgasmControl.groovy
+++ b/scripts/toy/orgasmControl.groovy
@@ -56,7 +56,7 @@
 				["OK, toy, I'm going to count you down.", "Would you like a countdown, toy?"],
 				["Get stroking!", "Jerk it!", "Stroke it for me!"]]);
 		if (mightCum(5 + getRandom(5))) return true;
-		present([nDRESSED, [TITS,PUSSY,ASS]], []);
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], []);
 		def mayCum = can(CUM) & !stateIs(CHASTE_TIMER) & sessionParams.askedCum;
 		def numbers = "";
 		def nmax = 10
@@ -70,7 +70,7 @@
 				return cumChanceDenied();
 			}
 		}
-		present([nDRESSED, [TITS,PUSSY,ASS]], [
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [
 				["Now! Cum", "Cum! Cum"],
 				["for me", "for your ${dommeTitle()}", ""],
 				["you little slut!", "my lucky toy!", "you dirty boy!"]]);
@@ -88,7 +88,7 @@
 	toy.metaClass.cumChanceEdge { probCum=20, probRuin=33 ->
 		def soft = getRandom(8) ? true : teaseSoften();
 		if (soft) if (!harden([[TITS,PUSSY,ASS]])) return false;
-		return edge(6, [nDRESSED, [TITS,PUSSY,ASS]], true, probCum, probRuin);
+		return edge(6, [nDRESSED, "$TITS|$PUSSY|$ASS"], true, probCum, probRuin);
 	};
 
 	toy.metaClass.cumChanceWindow { probCum=20, probRuin=33 ->
@@ -101,7 +101,7 @@
 		if (randRange(1,100) > probCum || !mayCum) {
 			return cumChanceDenied();
 		}
-		present([nDRESSED, [TITS,PUSSY,ASS]], [
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [
 				["Now! Cum", "Now! Try to cum", "Cum! Cum"],
 				["for me", "for your ${dommeTitle()}", ""],
 				["you little slut!", "my lucky toy!", "you dirty boy!"]]);
@@ -133,7 +133,7 @@
 			return cumChanceDenied();
 		}
 		for (def n = 3 + getRandom(6); n >= 0; n--) {
-			present([nDRESSED, [TITS,PUSSY,ASS]], [
+			present([nDRESSED, "$TITS|$PUSSY|$ASS"], [
 					["Now! Cum", "Now! Try to cum", "Cum! Cum"],
 					["for me", "for your ${dommeTitle()}", ""],
 					["you little slut!", "my lucky toy!", "you dirty boy!"]]);
@@ -143,7 +143,7 @@
 				return true;
 			}
 			if (n > 0) {
-				present([nDRESSED, [TITS,PUSSY,ASS]], [
+				present([nDRESSED, "$TITS|$PUSSY|$ASS"], [
 						["Stop!", "Not now!", "Wait!"],
 						["Hands off!", "Hands by your side."]]);
 				w = (w - 2) + getRandom(8);
--- a/scripts/toy/tease.groovy
+++ b/scripts/toy/tease.groovy
@@ -321,7 +321,7 @@
 		preRelease();
 		def soft = getRandom(8) ? true : teaseSoften();
 		if (soft) if (!harden([[TITS,PUSSY,ASS]])) return false;
-		strokes(15, [nDRESSED, [TITS,PUSSY,ASS]]);
+		strokes(15, [nDRESSED, "$TITS|$PUSSY|$ASS"]);
 		return true;
 	};
 
@@ -330,34 +330,34 @@
 		preRelease();
 		def soft = getRandom(8) ? true : teaseSoften();
 		if (soft) if (!harden([[TITS,PUSSY,ASS]])) return false;
-		edge(6, [nDRESSED, [TITS,PUSSY,ASS]]);
+		edge(6, [nDRESSED, "$TITS|$PUSSY|$ASS"]);
 		return true;
 	};
 
 	toy.metaClass.intTease {
 		if (!can(EDGE)) return false;
 		(getRandom(2) + 2).times {
-			speak(imageTagsComment(showImage([[TITS,PUSSY,ASS]])));
+			speak(imageTagsComment(showImage(["$TITS|$PUSSY|$ASS"])));
 			pause(getRandom(5) + 10);
 		};
-		present([[TITS,PUSSY,ASS]], [
+		present(["$TITS|$PUSSY|$ASS"], [
 				["You look so desperate.", "Can't hold back any more?", "Am I driving you mad?"]]);
 		pause(getRandom(10) + 5);
-		present([nDRESSED, [TITS,PUSSY,ASS]], [
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [
 				["How about I let you cum?", "Maybe I should let you cum.", "Do you want to cum for your ${dommeTitle()}?"],
 				["Would you like that?", "Let some of that frustration out?", "Would this make you happy?"]]);
 		pause(getRandom(10) + 5);
-		askToCum([nDRESSED, [TITS,PUSSY,ASS]]);
+		askToCum([nDRESSED, "$TITS|$PUSSY|$ASS"]);
 		pause(3);
 		return true;
 	};
 	
 	toy.metaClass.intTeaseBalls {
-		present([nDRESSED, [TITS,PUSSY,ASS]], [
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [
 				["Grab your balls and tap them", "Fondle your balls", "Massage your balls", "Grad your balls and trace a finger over them,", "Grab your balls and caress them"],
 				["ever so", ""], ["softly!", "gently!"]]);
 		pause(8 + getRandom(4));
-		present([nDRESSED, [TITS,PUSSY,ASS]], [["Keep going!", "Just like that!", "Ain't that nice?"]]);
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [["Keep going!", "Just like that!", "Ain't that nice?"]]);
 		pause(8 + getRandom(2));
 		return true;
 	};
@@ -366,14 +366,14 @@
 		if (!sessionToys[COCK]) return true;
 		def txt = ["Stroke the head of my cock with ${getRandom(2)?"one finger":"two fingers"}..."];
 		if (sessionToys[HARDON]) txt += ["Trace a finger along my cock, up and down... up and down...", "${getRandom(2)?"Rub":"Tap"} the head of my cock with one finger,"];
-		present([nDRESSED, [TITS,PUSSY,ASS]], [txt, ["ever so", ""], ["softly!", "gently!", "slowly!"]]);
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [txt, ["ever so", ""], ["softly!", "gently!", "slowly!"]]);
 		pause(8 + getRandom(4));
-		present([nDRESSED, [TITS,PUSSY,ASS]], [["Keep going!", "Just like that!", "Ain't that nice?", "Do you want more of that?"]]);
+		present([nDRESSED, "$TITS|$PUSSY|$ASS"], [["Keep going!", "Just like that!", "Ain't that nice?", "Do you want more of that?"]]);
 		pause(8 + getRandom(2));
 		return true;
 	};
 	
-	toy.metaClass.teaseNipples { imgSpec=[nDRESSED, [TITS,PUSSY,ASS]] ->
+	toy.metaClass.teaseNipples { imgSpec=[nDRESSED, "$TITS|$PUSSY|$ASS"] ->
 		if (!stateIs(NAKED) && !stateIs(CLAMPED)) {
 			present(imgSpec, [["Let me see", "I want to see", "Show me"], ["those", "your"], ["nipples!"]]);
 			pause(7);
--- a/scripts/toy/misc.groovy
+++ b/scripts/toy/misc.groovy
@@ -1,7 +1,7 @@
 { toy ->
 	toy.metaClass.intNothing {
 		(getRandom(3) + 2).times {
-			speak(imageTagsComment(showImage([[TITS,PUSSY,ASS]])));
+			speak(imageTagsComment(showImage(["$TITS|$PUSSY|$ASS"])));
 			pause(getRandom(10) + 10);
 		};
 		return true;
