--- a/scripts/toy.groovy
+++ b/scripts/toy.groovy
@@ -149,7 +149,7 @@
 	boolean has(String i) { (loadBoolean("toys.$i") == true) || (loadBoolean("toy.toys.$i") == true) || DEFAULTTOYS.contains(i) }
 	boolean likes(String i) { loadBoolean("fetish.$i") == true }
 	boolean can(String i) { loadBoolean("toy.permission.$i") == true }
-	boolean perm(String i) { can(it) }
+	boolean perm(String i) { can(i) }
 	boolean getPermission(String i) { getProp("permission.$i", loadB) }
 	void setPermission(String i, boolean v) { setProp("permission.$i", v) }
 	void givePermission(String i) { setPermission(i, true) }
