--- a/scripts/Domme/chores.groovy
+++ b/scripts/Domme/chores.groovy
@@ -13,7 +13,8 @@
 int chn = loadInteger("domme.chores.day")
 int knk = loadInteger("domme.kinky")
 int chw
-int chanceBenWaBalls = loadInteger("domme.chanceBenWaBalls")
+Integer chanceBenWaBalls = loadInteger("domme.chanceBenWaBalls")
+if(chanceBenWaBalls == null) chanceBenWaBalls = 8
 if(chanceBenWaBalls<8){
 	chanceBenWaBalls = 8
 	save("domme.chanceBenWaBalls",chanceBenWaBalls)
