--- a/scripts/Domme/punishment__sha256_cb9c2e31cf67.groovy
+++ b/scripts/Domme/punishment__sha256_cb9c2e31cf67.groovy
@@ -1473,7 +1473,7 @@
 		showButton("Thank you, Miss!")
 		break
 	case 5:
-		Show("Nothing?\n\nYou report for punishment finding out and telling me you don't need any?")
+		show("Nothing?\n\nYou report for punishment finding out and telling me you don't need any?")
 		wait(10)
 		show("Think before you act!")
 		wait(10)
