--- a/scripts/jewell/punishments.groovy
+++ b/scripts/jewell/punishments.groovy
@@ -643,7 +643,7 @@
 	break
 
 case "section81":
-	save("punishments.mood(",101-2)
+	save("punishments.mood",101-2)
 	save("cybermistress.points", 6+startPoints)
 	setImage("jewell/punishments.jpg")
 	show("(You've got "+loadInteger("cybermistress.points")+" points)\n"+"You must lay face down spread eagle on your bed with a dildo vibrating in your ass, only wearing panties for "+loadString("punishments.mood")+" minutes. Also tie your hands and legs to the bed.")
@@ -870,7 +870,7 @@
 	break
 
 case "section112":
-	save("punishments.random(",getRandom(3)+2)
+	save("punishments.random",getRandom(3)+2)
 	save("punishments.mood",101-2)
 	save("cybermistress.points", 6+startPoints)
 	setImage("jewell/punishments.jpg")
