--- a/scripts/DisciplineClinic__sha256_4d5c7e7be172.groovy
+++ b/scripts/DisciplineClinic__sha256_4d5c7e7be172.groovy
@@ -1883,8 +1883,8 @@
 	{
 		cornerPosPick = getRandom(cornerPosArray.size)+1
 		def cornerPosName = cornerPosArray[cornerPosPick]
-		if(positionName=="xxx") {
-			positionSelection = false	
+		if(cornerPosName=="xxx") {
+			cornerPosSelection = false	
 			setImageScold()
 			show(" ")
 			showButton(" ")
@@ -1892,8 +1892,8 @@
 			show(" ")
 			showButton("yes "+title+" ")
 		}
-		if(positionName=="xxx") {
-			positionSelection = false	
+		if(cornerPosName=="xxx") {
+			cornerPosSelection = false	
 			setImageScold()
 			show(" ")
 			showButton(" ")
--- a/scripts/DisciplineClinic__sha256_8a19a29ebfc8.groovy
+++ b/scripts/DisciplineClinic__sha256_8a19a29ebfc8.groovy
@@ -2508,8 +2508,8 @@
 	{
 		cornerPosPick = getRandom(cornerPosArray.size)+1
 		def cornerPosName = cornerPosArray[cornerPosPick]
-		if(positionName=="xxx") {
-			positionSelection = false	
+		if(cornerPosName=="xxx") {
+			cornerPosSelection = false	
 			setImageScold()
 			show(" ")
 			showButton(" ")
@@ -2517,8 +2517,8 @@
 			show(" ")
 			showButton("yes "+title+" ")
 		}
-		if(positionName=="xxx") {
-			positionSelection = false	
+		if(cornerPosName=="xxx") {
+			cornerPosSelection = false	
 			setImageScold()
 			show(" ")
 			showButton(" ")
--- a/scripts/DisciplineClinic__sha256_956b9beabe8f.groovy
+++ b/scripts/DisciplineClinic__sha256_956b9beabe8f.groovy
@@ -2494,8 +2494,8 @@
 	{
 		cornerPosPick = getRandom(cornerPosArray.size)+1
 		def cornerPosName = cornerPosArray[cornerPosPick]
-		if(positionName=="xxx") {
-			positionSelection = false	
+		if(cornerPosName=="xxx") {
+			cornerPosSelection = false	
 			setImageScold()
 			show(" ")
 			showButton(" ")
@@ -2503,8 +2503,8 @@
 			show(" ")
 			showButton("yes "+title+" ")
 		}
-		if(positionName=="xxx") {
-			positionSelection = false	
+		if(cornerPosName=="xxx") {
+			cornerPosSelection = false	
 			setImageScold()
 			show(" ")
 			showButton(" ")
