--- a/scripts/DisciplineClinic__sha256_4d5c7e7be172.groovy
+++ b/scripts/DisciplineClinic__sha256_4d5c7e7be172.groovy
@@ -1300,6 +1300,7 @@
 		dialog = dialogArray[getRandom(dialogArray.size)]
 		show(dialog)
 		wait(7)
+		if(lastTime==null) lastTime=0
 		elapsedTime = thisTime - lastTime
 		setImageReception()
 //	show("elapsedTime is: "+elapsedTime+"")
--- a/scripts/DisciplineClinic__sha256_8a19a29ebfc8.groovy
+++ b/scripts/DisciplineClinic__sha256_8a19a29ebfc8.groovy
@@ -1746,6 +1746,7 @@
 	if(countOffense > 1) {
 		setImageReception()
 		if(pickOffense==15) {		//maintenance
+			if(lastTime==null) lastTime=0
 			elapsedTime = thisTime - lastTime
 //			show("elapsedTime is: "+elapsedTime+"")
 //			showButton("")
@@ -1788,6 +1789,7 @@
 			dialog = dialogArray[getRandom(dialogArray.size)]
 			show(dialog)
 			wait(7)
+			if(lastTime==null) lastTime=0
 			elapsedTime = thisTime - lastTime
 			setImageReception()
 //	show("elapsedTime is: "+elapsedTime+"")
--- a/scripts/DisciplineClinic__sha256_956b9beabe8f.groovy
+++ b/scripts/DisciplineClinic__sha256_956b9beabe8f.groovy
@@ -1733,6 +1733,7 @@
 	if(countOffense > 1) {
 		setImageReception()
 		if(pickOffense==15) {		//maintenance
+			if(lastTime==null) lastTime=0
 			elapsedTime = thisTime - lastTime
 //			show("elapsedTime is: "+elapsedTime+"")
 //			showButton("")
@@ -1775,6 +1776,7 @@
 			dialog = dialogArray[getRandom(dialogArray.size)]
 			show(dialog)
 			wait(7)
+			if(lastTime==null) lastTime=0
 			elapsedTime = thisTime - lastTime
 			setImageReception()
 //	show("elapsedTime is: "+elapsedTime+"")
